module/zfs/zio.c

```diff
@@ -3278,6 +3331,24 @@ zio_write_gang_block(zio_t *pio, metaslab_class_t *mc)
 		zio_nowait(cio);
 	}
 
+	if (g > gbh_nblkptrs(SPA_OLD_GANGBLOCKSIZE)) {
+		gbh_tail(gbh, gangblocksize)->zgt_version = ZIO_GB_SIZED;
```

Comment:
It seems we never read or assert this version after setting.  I have some doubts about its usability.

Reply:
In theory, we could use this for debugging, but I struggle to think of a bug that could come up where it would be especially useful. I had some idea about future versions of gang blocks that could store a few bytes of data in the header itself to save an allocation, but we could always implement the version number when that gets implemented. I'll remove it from this PR.